File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
       if(s.length()!=t.length()){return false;}

       int n=s.length();
       int alpha[]=new int[26];
       for(int i=0;i<n;i++)
       {
        alpha[(int)(s.charAt(i)-'a')]++;
        alpha[(int)(t.charAt(i)-'a')]--;
       }
        
       for(int i=0;i<26;i++)
       {
        if(alpha[i]!=0)
        {
            return false;
        }
       }
       return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> map;

        if(s.size()!=t.size()){ return false;}

        for(int i=0;i<s.size();i++){
            map[s[i]]++;
            map[t[i]]--;
        }

        for(auto& both : map){
            if(both.second!=0){return false;}
        }

        return true;
    }
};